src/main/java/edu/si/fcrepo/AkubraIDConverter.java

```diff
@@ -0,0 +1,107 @@
+ * the License at: http://www.apache.org/licenses/
+ *
+ * This software and accompanying documentation is supplied without
+ * warranty of any kind. The copyright holder and the Smithsonian Institution:
+ * (1) expressly disclaim any warranties, express or implied, including but not
+ * limited to any implied warranties of merchantability, fitness for a
+ * particular purpose, title or non-infringement; (2) do not assume any legal
+ * liability or responsibility for the accuracy, completeness, or usefulness of
+ * the software; (3) do not represent that use of the software would not
+ * infringe privately owned rights; (4) do not warrant that the software
+ * is error-free or will be maintained, supported, updated or enhanced;
+ * (5) will not be liable for any indirect, incidental, consequential special
+ * or punitive damages of any kind or nature, including but not limited to lost
+ * profits or loss of data, on any basis arising from contract, tort or
+ * otherwise, even if any of the parties has been warned of the possibility of
+ * such loss or damage.
+ *
+ * This distribution includes several third-party libraries, each with their own
+ * license terms. For a complete copy of all copyright and license terms, including
+ * those of third-party libraries, please see the product release notes.
+ */
+
+package edu.si.fcrepo;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+
+/**
+ * Utility functions extracted/modified based on org.fcrepo.server.storage.lowlevel.akubra.AkubraLowLevelStorage.
+ * Original code authored by Chris Wilper
+ *
+ * @author whikloj
+ */
+public class AkubraIDConverter {
+
+    /**
+     * Rather than bring in the entire PID class to get a string.
+     */
+    public static String FEDORA_URI = "info:fedora/";
+
+    /**
+     * Converts a token to a token-as-blobId.
+     *
+     * @param token the token to convert
+     * @return the AkubraLowLevelStorage BlobId
+     */
+    public static URI getBlobId(final URI token) {
+        return getBlobId(token.toString());
+    }
+
+    /**
+     * Converts a token to a token-as-blobId.
+     * <p>
+     * Object tokens are simply prepended with <code>info:fedora/</code>, whereas datastream tokens are additionally
+     * converted such that <code>ns:id+dsId+dsVersionId</code> becomes
+     * <code>info:fedora/ns:id/dsId/dsVersionId</code>, with the dsId and dsVersionId segments URI-percent-encoded
+     * with UTF-8 character encoding.
+     *
+     * @param token the token to convert.
+     * @return the blob id.
+     * @throws IllegalArgumentException if the token is not a well-formed pid or datastream token.
+     */
+    public static URI getBlobId(final String token) {
+        try {
+            final int i = token.indexOf('+');
+            if (i == -1) {
+                if (token.startsWith(FEDORA_URI)) {
```

Comment:
I'm still a bit weirded out by this. Do you have any theory as to how your repo ended up with URIs lacking the prefix? I can't remember ever having seen that before.

Reply:
Also, I'd prefer a `?:` here instead of a `if-else`.

Reply:
I tried creating a new object with a  new Fedora 3.8.1 repo and adding a datastream (all using the Flash admin tool) and it does not persist the "info:fedora/" prefix when referencing any managed datastreams. Only thing I can think is
```
<foxml:contentLocation TYPE="INTERNAL_ID" REF="uofm:2553586+OBJ+OBJ.0"/>
```
are there other `TYPE`s that might use the `info:fedora/` prefix?